Framework/DataHandling/test/LoadInstrumentTest.h

```diff
@@ -119,14 +106,15 @@ class LoadInstrumentTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(samplepos->getName(), "nickel-holder");
     TS_ASSERT_DELTA(samplepos->getPos().Z(), 0.0, 0.01);
 
-    boost::shared_ptr<const IDetector> ptrDet103 = i->getDetector(103);
-    TS_ASSERT_EQUALS(ptrDet103->getID(), 103);
-    TS_ASSERT_EQUALS(ptrDet103->getName(), "pixel");
-    TS_ASSERT_DELTA(ptrDet103->getPos().X(), 0.4013, 0.01);
-    TS_ASSERT_DELTA(ptrDet103->getPos().Z(), 2.4470, 0.01);
-    double d = ptrDet103->getPos().distance(samplepos->getPos());
+    const auto &detectorInfo = output->detectorInfo();
+    const auto &ptrDet103 = detectorInfo.detector(103);
```

Comment:
This makes the asserts below fail. For this to pass I have to use detector `102`.